src/main/java/com/minecolonies/colony/permissions/Permissions.java

```diff
@@ -552,188 +505,233 @@ public void serializeViewNetworkData(ByteBuf buf, Permissions.Rank viewerRank)
         }
 
+        @NotNull
         public Set<Player> getPlayersByRank(Rank rank)
         {
             return Collections.unmodifiableSet(
-                    this.players.values()
-                            .stream()
-                            .filter(player -> player.rank == rank)
-                            .collect(Collectors.toSet()));
+              this.players.values()
+                .stream()
+                .filter(player -> player.rank == rank)
+                .collect(Collectors.toSet()));
         }
 
-        public Set<Player> getPlayersByRank(Set<Rank> ranks)
+        @NotNull
+        public Set<Player> getPlayersByRank(@NotNull Set<Rank> ranks)
         {
             return Collections.unmodifiableSet(
-                    this.players.values()
-                            .stream()
-                            .filter(player -> ranks.contains(player.rank))
-                            .collect(Collectors.toSet()));
+              this.players.values()
+                .stream()
+                .filter(player -> ranks.contains(player.rank))
+                .collect(Collectors.toSet()));
         }
 
+        @NotNull
         public Map<Rank, Integer> getPermissions()
         {
             return permissions;
         }
 
-        public Rank getRank(EntityPlayer player)
-        {
-            return getRank(player.getUniqueID());
-        }
-
-        @Override
-        public Rank getRank(UUID id)
-        {
-            Player player = players.get(id);
-            return player != null ? player.rank : Rank.NEUTRAL;
-        }
-
-        @Override
-        public boolean hasPermission(EntityPlayer player, Action action)
-        {
-            return hasPermission(getRank(player), action);
-        }
-
-        @Override
-        public boolean isColonyMember(EntityPlayer player)
-        {
-            return players.containsKey(player.getUniqueID());
-        }
-
-        public boolean hasPermission(UUID id, Action action)
+        public boolean hasPermission(UUID id, @NotNull Action action)
         {
             return hasPermission(getRank(id), action);
-        }
-
-        public boolean hasPermission(Rank rank, Action action)
+        }        @NotNull
+    public Rank getRank(@NotNull EntityPlayer player)
         {
-            return Utils.testFlag(permissions.get(rank), action.flag);
+            return getRank(player.getUniqueID());
         }
 
-        public boolean setPermission(Rank rank, Action action)
+        public boolean setPermission(Rank rank, @NotNull Action action)
         {
             int flags = permissions.get(rank);
 
             //check that flag isn't set
-            if(!Utils.testFlag(flags, action.flag))
+            if (!Utils.testFlag(flags, action.flag))
             {
                 permissions.put(rank, Utils.setFlag(flags, action.flag));
                 return true;
             }
             return false;
+        }        @NotNull
+    @Override
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Annotation 'Override' have incorrect indentation level 4, expected level should be 17. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.annotation.AnnotationLocationCheck)